plugins/telescope/file-browser.nix

```diff
@@ -0,0 +1,188 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.telescope.extensions.file_browser;
+  helpers = import ../helpers.nix {inherit lib;};
+
+  hiddenOption = types.submodule {
+    options = {
+      fileBrowser = helpers.mkNullOrOption types.bool "";
+
+      folderBrowser = helpers.mkNullOrOption types.bool "";
+    };
+  };
+
+  statOption = types.submodule {
+    options = {
+      width = helpers.mkNullOrOption types.int "";
+      display = helpers.mkNullOrOption helpers.rawType "";
+      hl = helpers.mkNullOrOption types.str "";
+    };
+  };
+in {
+  options.plugins.telescope.extensions.file_browser = {
+    enable = mkEnableOption "File browser extension for telescope";
+
+    package = helpers.mkPackageOption "telescope file_browser" pkgs.vimPlugins.telescope-file-browser-nvim;
+
+    theme = helpers.mkNullOrOption types.str "Custom theme, will use your global theme by default";
+
+    path = helpers.defaultNullOpts.mkNullable (types.either types.str helpers.rawType) "vim.loop.cwd()" ''
+      dir to browse files from, `vim.fn.expanded` automatically
+    '';
+
+    cwd = helpers.defaultNullOpts.mkNullable (types.either types.str helpers.rawType) "vim.loop.cwd()" ''
+      dir to browse folders from, `vim.fn.expanded` automatically
+    '';
+
+    cwdToPath = helpers.defaultNullOpts.mkBool false ''
+      whether folder browser is launched from `path` rather than `cwd`
+    '';
+
+    grouped = helpers.defaultNullOpts.mkBool false ''
+      group initial sorting by directories and then files
+    '';
+
+    files = helpers.defaultNullOpts.mkBool true ''
+      start in file (true) or folder (false) browser
+    '';
+
+    addDirs = helpers.defaultNullOpts.mkBool true ''
+      whether the file browser shows folders
+    '';
+
+    depth = helpers.defaultNullOpts.mkInt 1 ''
+      file tree depth to display, `false` for unlimited depth
+    '';
+
+    autoDepth = helpers.defaultNullOpts.mkBool false ''
+      unlimit or set `depth` to `auto_depth` & unset grouped on prompt for file_browser
+    '';
+
+    selectBuffer = helpers.defaultNullOpts.mkBool false ''
+      select current buffer if possible; may imply `hidden=true`
+    '';
+
+    hidden = helpers.defaultNullOpts.mkNullable (types.either types.bool hiddenOption) "false" ''
+      determines whether to show hidden files or not
+    '';
+
+    respectGitignore = helpers.defaultNullOpts.mkBool false ''
+      induces slow-down w/ plenary finder (true if `fd` available)
+    '';
+
+    browseFiles = helpers.defaultNullOpts.mkNullable helpers.rawType "fb_finders.browse_files" ''
+      a custom lua function to override for the file browser
+    '';
+    browseFolders = helpers.defaultNullOpts.mkNullable helpers.rawType "fb_finders.browse_folders" ''
+      a custom lua function to override for the folder browser
+    '';
+
```

Comment:
If some lua code is expected, we usually use a `str` option and we automatically perform the conversion to `rawType` in the implementation.

Reply:
See [this example](https://github.com/pta2002/nixvim/blob/4a9a3c1f6cea52d756d406d6467b48898cdbff07/plugins/utils/nvim-ufo.nix#L63) for instance.